### PR TITLE
WP-82 - Added additional models

### PR DESCRIPTION
**Pull request by usersblock**
Added sublist,movie,follow and modified user to prevent table pluralizing.

### CODE REVIEW

File: WatchParty/backend/models/sublist.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module.exports = (sequelize, DataTypes) => {
+  const Sublist = sequelize.define("Sublist", {
+    name: {
+		type: DataTypes.STRING(32),
+		allowNull: false,
+		unique: 'uniquecombination',
```
Comment (PredatorFeesh) [inline WatchParty/backend/models/sublist.js]: I think we also need to add fields https://stackoverflow.com/a/42199137

File: WatchParty/backend/models/movie.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      validate: { isEmail: true }
+    },
+    tmdbid: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      validate: { notNull: true }
+    },
+    watchstate: {
+      type: DataTypes.ENUM,
+	  values:['watched','to-watch','not-interested'],
```
Comment (PredatorFeesh) [inline WatchParty/backend/models/movie.js:17]: Beautiful 🚀

**Review by PredatorFeesh — COMMENTED** [review]
One thing I noticed; we don't define `userid` in models. I know the ID is auto generated, but is this the name it's given?

**Comment by usersblock** [comment]
I believe it's only called id, but for the sake of differentiating what belongs to what I named it userid.

File: WatchParty/backend/models/sublist.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+},{
+    freezeTableName: true
+	});
+	Sublist.associate = function(models) {
+	    Sublist.belongsTo(models.User,{
+		foreignKey: 'userid',
+		onDelete: 'cascade'
+	});
+	Sublist.belongsTo(models.Movie,{
```
Comment (PredatorFeesh) [inline WatchParty/backend/models/sublist.js:19]: This is a one-to-many relation (we have 1 sublist for many movies https://sequelize.readthedocs.io/en/latest/docs/associations/#one-to-many-associations)

File: WatchParty/backend/models/follow.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    follower: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      validate: { notNull: true }
+    }
+},{
+    freezeTableName: true
+	});
+	Follow.associate = function(models) {
```
Comment (PredatorFeesh) [inline WatchParty/backend/models/follow.js:18]: These two are belongs-to-many (you can specify how  here https://sequelize.readthedocs.io/en/latest/docs/associations/#belongs-to-many-associations )

File: WatchParty/backend/models/sublist.js (hunk 3 above)
Comment (usersblock) [reply WatchParty/backend/models/sublist.js:19]: Added has many to user and movie

File: WatchParty/backend/models/follow.js (hunk 4 above)
Comment (usersblock) [reply WatchParty/backend/models/follow.js:18]: Changed both to belongs-to-many.

**Review by ian-s-mcb — CHANGES_REQUESTED** [review]
It would be a good idea to run the backend server before submitting this PR.

Sequelize throws a bunch of errors. I was able to fix them by removing the duplicate table fields. When you specify an associated key in the options object, Sequelize automatically adds those fields to the table, so specifying them again in the schema object causes a duplicate field error. The other error I noticed was that the Sublist model doesn't need an association with the User model because Sublist already has an association with Movie, which provides an association to User. Directly adding an association to User causes a duplicate field error.

**Comment by usersblock** [comment]
Belongstomany creates a new table to hold relations and  hasmany adds a foreign key to the target. Belongsto is relatively simple as it just connects a foreignkey to another table. To keep each parameter in each model file, I think we should only use Belongsto. As for sublist, I've removed the userid attribute and had unique be applied to name and movie. I've added the change to dbnotes as well.

**Comment by PredatorFeesh** [comment]
For BelongsToMany, you can use a `through` table. Whichever way you think it's best done, we'll go with that as long as it can pass the tests and edge cases

**Review by PredatorFeesh — APPROVED** [review]

**Review by ian-s-mcb — APPROVED** [review]
